pandas/core/indexes/base.py

```diff
@@ -1154,10 +1158,17 @@ def to_frame(self, index=True):
         0   Ant
         1  Bear
         2   Cow
+
+        >>> idx.to_frame(index=False, name='zoo')
+            zoo
+        0   Ant
+        1  Bear
+        2   Cow
         """
 
         from pandas import DataFrame
-        name = self.name or 0
+        if not name:
```

Comment:
This would have to be if name is not None, to allow falsey names.

Reply:
You are right! I will change that